requests/src/main/scala/weco/api/requests/services/ItemLookup.scala

```diff
@@ -1,21 +1,146 @@
+
+    elasticsearchService.findBySearch[Work[Indexed]](searchRequest).map {
+      case Left(err) => Left(err)
+      case Right(works) =>
+        val item =
+          works
+            .flatMap { _.data.items }
+            .collectFirst {
+              case item @ Item(IdState.Identified(id, _, _), _, _, _)
+                  if id == itemId =>
+                // This .asInstanceOf[] is a no-op to help the compiler see what
+                // we can see by reading the code.
+                item.asInstanceOf[Item[IdState.Identified]]
+            }
+
+        item match {
+          case Some(it) => Right(it)
+          case None     => Left(DocumentNotFoundError(itemId))
+        }
+    }
+  }
 
   def bySourceIdentifier(
     sourceIdentifiers: Seq[SourceIdentifier]
-  ): Future[Seq[Either[ElasticsearchError, Item[IdState.Identified]]]]
+  ): Future[Seq[Either[ElasticsearchError, RequestedItemWithWork]]] = {
+    val multiSearchRequest = MultiSearchRequest(
+      sourceIdentifiers.map { sourceIdentifier =>
+        search(index)
+          .query(
+            boolQuery
+              .must(
+                termQuery(
+                  field = "type",
+                  value = "Visible"
+                ),
+                termQuery(
+                  field = "data.items.id.sourceIdentifier.value",
+                  value = sourceIdentifier.value
+                )
+              )
+          )
+          .sortBy(
+            fieldSort("state.sourceIdentifier.value")
+              .order(SortOrder.Asc)
+          )
+          // We are sorting so we're only interested in the top value
+          .size(1)
+      }
+    )
+
+    elasticsearchService
+      .findByMultiSearch[Work[Indexed]](multiSearchRequest)
+      .map {
+        _.zip(sourceIdentifiers).map {
+          case (Right(works), srcId) =>
+            works
+              .flatMap { work =>
+                work.data.items.map(
+                  item => (work.data.title, work.state.canonicalId, item)
+                )
+              }
+              .collect {
+                case (
+                    title,
+                    workId,
+                    item @ Item(id @ IdState.Identified(_, _, _), _, _, _)
+                    ) if id.sourceIdentifier == srcId =>
+                  // This .asInstanceOf[] is a no-op to help the compiler see what
+                  // we can see by reading the code.
+                  item.asInstanceOf[Item[IdState.Identified]]
+                  RequestedItemWithWork(
+                    workId = workId,
+                    workTitle = title,
+                    item = item.asInstanceOf[Item[IdState.Identified]]
+                  )
+              }
+              .toList match {
+              // We can return multiple items from multiple works
+              // The sortBy above returns us the lowest bibId work association
+              case List(item) => Right(item)
+              case item :: _  => Right(item)
```

Comment:
Are these cases not the same? ie I think `List(item) == item :: Nil`

Reply:
I think this case is for the list that has more than one item.

Reply:
Right, so I think we just need the 2nd line (which will perform the work of the first too)? ie `case item :: _  => Right(item)`

Reply:
Yeah, I see what you mean. I'll change this.

Reply:
"take the first item of a non-empty list"